README.md

```diff
@@ -22,12 +33,105 @@ It supports TensorFlow, PyTorch, MXNet, and XGBoost on Python 3.6+.
+```
+
+#### Zero Script Change
+
+You can use your own training script while using [AWS Deep Learning Containers (DLC)](https://aws.amazon.com/machine-learning/containers/) in TensorFlow, PyTorch, MXNet and XGBoost frameworks. The AWS DLCs enable you to use Debugger with no changes to your training script by automatically adding SageMaker Debugger's `Hook`.
+The following table shows currently supported versions of the four frameworks for Zero Script Change experience.
+
+| Framework | Version |
+| --- | --- |
+| [TensorFlow](tensorflow.md) | 1.15, 2.1 |
+| [MXNet](mxnet.md) | 1.6 |
+| [PyTorch](pytorch.md) | 1.3, 1.4 |
+| [XGBoost](xgboost.md) | >=0.90-2 [As Built-in algorithm](xgboost.md#use-xgboost-as-a-built-in-algorithm)|
+
+
+#### Bring your own training container
+
+| Framework | Versions |
+| --- | --- |
+| [TensorFlow](tensorflow.md) | 1.14. 1.15, 2.0.1, 2.1.0 |
+| Keras (with TensorFlow backend) | 2.3 |
+| [MXNet](mxnet.md) | 1.4, 1.5, 1.6 |
+| [PyTorch](pytorch.md) | 1.2, 1.3, 1.4 |
+| [XGBoost](xgboost.md) | [As Framework](xgboost.md#use-xgboost-as-a-framework) |
+
+#### Support for Distributed Training and Known Limitations
+
+<table>
+    <thead>
+        <tr>
+           <th colspan=3>
+           Distributed Training
+           </th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan=2>Horovod</td>
+            <td>Supported</td>
+            <td>TF 1.15, PT 1.4, MX 1.6</td>
+        </tr>
+        <tr>
+            <td>Not supported</td>
+            <td>TF 2.x, PT 1.5</td>
+        </tr>
+        <tr>
+            <td>Parameter Server-based</td>
+            <td colspan=2>Not supported</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## How It Works
+
+Amazon SageMaker Debugger uses the construct of a `Hook` to save the values of requested tensors throughout the training process. You can then setup a `Rule` job which simultaneously monitors and validates these tensors to ensure
+that training is progressing as expected.
+A rule might check for vanishing gradients, or exploding tensor values, or poor weight initialization. Rules are attached to CloudWatch events, so that when a rule is triggered it changes the state of the CloudWatch event. You can configure any action on the CloudWatch event, such as to stop the training job saving you time and money.
+
+Amazon SageMaker Debugger can be used inside or outside of SageMaker. However the built-in rules that AWS provides are only available for SageMaker training. Scenarios of usage can be classified into the following:
+- **SageMaker Zero Script Change**: Here you specify which rules to use when setting up the estimator and run your existing script without no change. For an example of how to  [Running a Rule with Zero Script Change on SageMaker](#running-a-rule-with-zero-script-change-on-sageMaker).
+- **SageMaker Bring Your Own Container**: Here you specify the rules to use and modify your training script minimally to enable SageMaker Debugger.
```

Comment:
this requires a dedicated page. explaining step by step, how can i bring my own container.

Reply:
As discussed above, let me know if you can provide a piece of actual code.

Reply:
Current solution: linking to the markdown doc files for the four frameworks.
Going to review those files.

Reply:
Added one of the new BYOC snippet of code using TF 2.x GradientTape from tensorflow.md